special-pages/pages/new-tab/app/omnibar/components/Container.js

```diff
@@ -0,0 +1,34 @@
+import { h } from 'preact';
+import styles from './Container.module.css';
+import { useRef, useEffect, useState } from 'preact/hooks';
+
+/**
+ * @param {object} props
+ * @param {boolean} props.overflow
+ * @param {import('preact').ComponentChildren} props.children
+ */
+export function Container({ overflow, children }) {
+    const contentRef = useRef(/** @type {HTMLDivElement|null} */ (null));
+    const initialHeight = useRef(/** @type {number|null} */ (null));
+    const [contentHeight, setContentHeight] = useState(/** @type {number|null} */ (null));
+
+    useEffect(() => {
+        const content = contentRef.current;
+        if (!content) return;
+
+        initialHeight.current = content.scrollHeight;
+        setContentHeight(content.scrollHeight);
+
+        const resizeObserver = new ResizeObserver(() => setContentHeight(content.scrollHeight));
+        resizeObserver.observe(content);
+        return () => resizeObserver.disconnect();
+    }, []);
```

Comment:
this should be a useLayoutEffect here since you're using the height calculation in the rendering